### PR TITLE
[Accordion] Create accordion component

### LINKED ISSUE #583: [Accordion] Create Accordion component
Hi

I really like Core Components idea and ways of working.
In our project we need to deliver Accordion component,  when we look at Core Components we can see very similar component: Tabs. There is only one feature missing, Tabs component doesn’t have an option for managing panels visibility like all-open and all-closed. Tabs have only n-th open.

What is your recommendation in deliver Accordion Component?

CQ-4256161

### PR DESCRIPTION
**Pull request by richardhand**
Builds on the fantastic initial implementation by @brandonmaynard in #600 with some added functionality, ui-js e2e test case coverage, and a component library example page. Also includes some refactoring of the markup, OOTB CSS and ui widget. 

### Features
* Allows one or more initially expanded items to be set.
* Allows single item expansion to be enabled (in which case, one accordion item is forced to be open at a time).
* Allows definition of the accordion heading elements, if enabled via the content policy. Merges the `AllowedTitleSizesDataSourceServlet` (title component specific) into a new generic `AllowedHeadingElementsDataSourceServlet` for providing heading element options in an edit dialog based on content policy selection.
* Supports the panel selector mechanism to allow switching between accordion items in edit mode.
* Supports the childrenEditor item editing in the edit dialog.
* UI widget with simple user interaction for expanding/collapsing panels and keyboard accessibility (down/right, up/left, home/end, space/enter).

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #583` 
| Patch: Bug Fix?          |
| Minor: New Feature?      | New component
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

**Comment by richardhand** [comment]
@brandonmaynard, @jakublamprecht you might want to take a look at this one too before we merge?

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Accordion.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     *
+     * @return true if the accordion forces a single item to be expanded at a time; false otherwise
+     * @since com.adobe.cq.wcm.core.components.models 12.8.0
+     */
+    default boolean isSingleExpansion() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns the expanded items.
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Accordion.java]: I would mention by default, e.g.: 
Returns the items that are expanded by default.

File: content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+* Item header HTML element is configurable (`h2` - `h6`).
+* Editing features for accordion items (adding, removing, editing, re-ordering).
+
+### Use Object
+The Accordion component uses the `com.adobe.cq.wcm.core.components.models.Accordion` Sling model as its Use-object.
+
+### Component Policy Configuration Properties
+The following configuration properties are used:
+
+1. `./allowedHeadingElements` - the heading elements (`h2` - `h6`) that are allowed to be selected in the edit dialog.
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/README.md]: Shouldn't `h1` be also allowed?

File: content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/editor/js/accordion.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+            expandedSelectField.setDisabled(true);
+            expandedSelectSingleField.setDisabled(false);
+            expandedSelectSingleToggleable.show();
+        } else {
+            expandedSelectSingleToggleable.hide();
+            expandedSelectSingleField.setDisabled(true);
+            expandedSelectField.setDisabled(false);
+            expandedSelectToggleable.show();
+        }
+    }
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/editor/js/accordion.js]: For better readability, I'd rewrite this block as follows:
```
        if (hideAll) {
            expandedSelectField.setDisabled(true);
            expandedSelectToggleable.hide();
            expandedSelectSingleField.setDisabled(true);
            expandedSelectSingleToggleable.hide();
        } else if (singleExpansion) {
            expandedSelectField.setDisabled(true);
            expandedSelectToggleable.hide();
            expandedSelectSingleField.setDisabled(false);
            expandedSelectSingleToggleable.show();
        } else {
            expandedSelectField.setDisabled(false);
            expandedSelectToggleable.show();
            expandedSelectSingleField.setDisabled(true);
            expandedSelectSingleToggleable.hide();
        }
```

File: content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js [hunk 4]
```diff
@@ -0,0 +1,512 @@
...
+
+    var dataAttributes = {
+        item: {
+            expanded: "data-cmp-expanded"
+        }
+    };
+
+    var properties = {
+        /**
+         * Determines whether a only single accordion item is forced to be expanded at a time.
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js]: I would remove `only`.

**Comment by richardhand** [comment]
Thanks @jckautzmann - addressed your feedback in ab36528584365a5.

**Comment by jakublamprecht** [comment]
Hey @richardhand, sorry I missed that PR, but it looks good to me :)

File: content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/accordion.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+<div data-sly-use.accordion="com.adobe.cq.wcm.core.components.models.Accordion"
+     class="cmp-accordion"
+     data-cmp-is="accordion"
+     data-cmp-single-expansion="${accordion.singleExpansion}">
+    <div data-sly-test="${accordion.items && accordion.items.size > 0}"
```
Comment (bpauli) [inline content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/accordion.html]: accordion.items is NotNull so a test accordion.items.size > 0 should be sufficient.

Comment (richardhand) [reply content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/accordion.html]: Thanks @pauli, added in e5bda1228b328195f63ef2f215675aa2bb177ccb

**Comment by brandonmaynard** [comment]
I'm sorry @richardhand, I didn't see that you tagged me on this until today. For what its worth, your changes look good even though this has been merged already. My apologies

**Pull request merged**